Add Postgres tasks

zangell44 opened the pull request:
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] updates `CHANGELOG.md` (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?

This PR adds two new tasks for interacting with Postgres databases, `PostgresExecute` and `PostgresFetch`, which execute and fetch results of a SQL query.

Basic tests have been added for both tasks. Functionality has been tested locally with a live Postgres instance.

## Why is this PR important?

Closes [#1022](https://github.com/PrefectHQ/prefect/issues/1022).

cicdw commented on src/prefect/tasks/postgres/postgres.py:
```diff
@@ -0,0 +1,200 @@
+            - db_name (str):
+            - user (str):
+            - password (str):
+            - host (str):
```
Forgotten arg docs?

zangell44 replied:
Damnit haha I can't believe I forgot those

cicdw commented on src/prefect/tasks/postgres/postgres.py:
```diff
@@ -0,0 +1,200 @@
+            - db_name (str):
+            - user (str):
+            - password (str):
+            - host (str):
```
Same as above

zangell44 commented:
@cicdw yes, I can add that markdown file.

I noticed that the API References in the current ones link to 404 pages. Should I follow the same format for the links and expect those pages to be functional?

cicdw commented:
@zangell44 where are you seeing 404 errors?  Ultimately those links get deployed here: https://docs.prefect.io/guide/task_library/aws.html (for example) and these links should work.

zangell44 commented:
@cicdw gotcha, I was seeing 404 errors from when you click on the API references directly in the markdown files here https://github.com/PrefectHQ/prefect/blob/master/docs/guide/task_library/aws.md

I'll push changes to add a markdown file in a bit. 

I also came across the SQLite tasks and I think it's better to rename the ones I created to something like `PostgresExecute` and `PostgresFetch` to be more consistent/descriptive.

joshmeek commented on src/prefect/tasks/postgres/postgres.py:
```diff
@@ -0,0 +1,200 @@
+        Returns:
+            - None
+        """
+        if query == "":
```
Let's default the `query` to None and change this to `if not query` so it'll catch the cases where it's None and ""

joshmeek commented on src/prefect/tasks/postgres/postgres.py:
```diff
@@ -0,0 +1,200 @@
+
+        Returns:
+            - None
+        """
```
Can you add to the docstring

```
Raises:
	- DatabaseError: ..why it would be raised...
```

joshmeek replied:
Not sure what to do about the base `Exception` raise option though. Does that needs to be documented @cicdw?

joshmeek replied:
Oh, can you also add one for the `ValueError` as well?

cicdw replied:
I don't think the base `Exception` needs to be documented; I view the `Raises` section as the errors that one might "expect" to see when running these tasks and those that are explicitly raised by the task.

joshmeek commented on src/prefect/tasks/postgres/postgres.py:
```diff
@@ -0,0 +1,200 @@
+        Returns:
+            - records (tuple or list of tuples): records from provided query
+        """
+        if query == "":
```
Same query comment as the other task

joshmeek commented on src/prefect/tasks/postgres/postgres.py:
```diff
@@ -0,0 +1,200 @@
+
+        Returns:
+            - records (tuple or list of tuples): records from provided query
+        """
```
Same `Raises` docstring requests as the other task

zangell44 commented:
@joshmeek I added the exceptions and modified the logic for the query parameter in the latest commits. Let me know if there's anything else I can do on this.

Also, a quick GitHub etiquette thing - would you prefer I "Resolve conversation" after making requested edits, or leave the conversation until you review changes?

joshmeek commented:
@zangell44 Looks great! I'm all fine with marking the convos as resolved once you implement them

cicdw commented:
I prematurely merged my own PR and didn't think about the merge conflict this would cause, my apologies!!
